IdentityCore/src/util/ASAuthorizationSingleSignOnProvider+MSIDExtensions.m

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "ASAuthorizationSingleSignOnProvider+MSIDExtensions.h"
+
+@implementation ASAuthorizationSingleSignOnProvider (MSIDExtensions)
+
++ (ASAuthorizationSingleSignOnProvider *)msidSharedProvider
+{
+    static dispatch_once_t once;
+    static ASAuthorizationSingleSignOnProvider *ssoProvider;
+    
+    dispatch_once(&once, ^{
+        NSURL *url = [NSURL URLWithString:@"https://login.microsoftonline.com"];
```

Comment:
nit: use constant for this one, as we should already have it declared and using in other places.

Reply:
done